### ISSUE TITLE
improve data tests to do multiple values

### ISSUE THREAD

**Issue by huss** [OpenEnergyDashboard/OED#698]
### Is your feature request related to a problem? Please describe.

There are many places that OED tests that a certain value matches the expected value. While this is a good test, it would be better if it was done for multiple values. Examples include:

- src/server/test/routes/compressedReadingsRouteTests.js seems to test a single value.
- src/server/test/db/readingTests.js tests 1 or 2 points
- src/server/test/db/newCompressedReadingsTests.js tests limited points
- src/server/test/db/compareTests.js does a couple of points
- src/server/test/db/barchartReadingsTests.js does a few points

These tests also tend to have readings with 1 hour length.

### Describe the solution you'd like

OED now allows for readings of arbitrary length. Thus, tests should try reading lengths other than 1 hour. The automated test data (see src/server/data/automatedTestingData.js) includes 23 minute data for this reason. The other reason 23 minutes was chosen is that readings cross the hour and day boundaries so are better tests of the compression algorithms. We should be using automated tests to do this type of testing.

Since we can generate multiple points, it would be better to loop over a number of points and test each one in an automated fashion rather than doing a few specialized points.

The graphing is now done with compressed data that can be daily, hourly and original (raw) meter data. We should be testing the values returned from these to be sure they are correct (already done to some extent).

We should more generally test that groups properly add the included meters. The automated data for sine^2 + cosine^2 that gives a constant value is a good example of how this could be done.

### Describe alternatives you've considered

We can continue to test limited number of points and for easy time frames for those points and do the rest manually but that is not what we want.

### Additional context

None.

**Comment by huss** [comment]
The [testing designDocs](https://github.com/OpenEnergyDashboard/DesignDocs/blob/main/testing/testing.md) gives more tests to be added to src/server/routes/readings.js. At this time they are line and bar reading tests with more to come. The section "Expected values" has tables for line and bar tests that indicate which have yet to be done. Please communicate with the OED project before starting these tests so duplicate efforts are not made. Note this is a reserved tagged issue since it is normally done by certain students and not for others to complete.
